### ISSUE TITLE
Provide guidance on including a hostname in a uriBaseIdValue

### ISSUE THREAD

**Issue by lgolding** [oasis-tcs/sarif-spec#113]
A URI base id might or might not include a hostname component:
```
{
  "runs": [
    {
      "originalUriBaseIds": {
        "SRCROOT": "file:///C:/src",           # Without a hostname
        "BINROOT": "file://LARRY-DEV/C:/bin",  # With a hostname
      }
    }
  ]
}
```
Should the spec provide guidance on whether or not to do that?

**Comment by lgolding** [comment]
IMO we should advise against it. Omitting the hostname allows relative URIs to be immediately resolved (that is, without having to rebase them) on any machine with the same file system layout as the original producer -- including the original producer's machine itself. Including the hostname means that the URI can be immediately resolved _only_ on the original producer's machine, and users on all other machines will have to rebase. (For example, viewers on other machines would have to prompt the user for the right location.)

**Comment by michaelcfanning** [comment]
I think we should recommend that users include it. Consumers can always disregard this information if it's helpful in the scenario you cite.

**Comment by michaelcfanning** [comment]
Producers should emit the host name. Post-processers may elide this information or inject it, depending on what they know about the larger results mgmt. system and its requirements.

**Comment by michaelcfanning** [comment]
suggest calling this uriBaseIdValues

**Comment by lgolding** [comment]
@michaelcfanning re: your suggestion to rename the property `originalUriBaseIds` to `uriBaseIdValues`: IMO the word "values" in an identifier adds no information. We might just as well call the property `uriBaseIds` -- and then my question is, "_Which_ URI base ids?"

I think the name we have is good.

**Comment by lgolding** [comment]
@michaelcfanning Do you want this guidance to apply _only_ to URIs that appear within `run.originalUriBaseIds`? For example:
```
{                                      # A run object.
"originalUriBaseIds": {
  "SRCROOT": "file:///larry-dev.example.com/C:/src/"
}
```
Or do you also want it to apply to any absolute URI that appear in `fileLocation` objects? For example:
```
{                                      # A result object.
  "analysisTarget": {
    "uri": "file:///larry-dev.example.com/C:/src/"
  }
}
````

**Comment by lgolding** [comment]
Also, what do you want to recommend for absolute URIs that _do_ omit the hostname (for example, what do you want the post-processor to omit)? There are two options, according to [RFC 8089](https://tools.ietf.org/html/rfc8089):
1. Simply omit the `file-auth` portion of the syntax:
    `file:///C:/src/`

2. Use the `local-path` syntax:
    `file:/C:/src`

In my draft, I recommended ("**SHOULD**") Option 2.

**Comment by lgolding** [comment]
Closed by 0121635.
